packages/beav.rb

```diff
@@ -3,50 +3,30 @@
 class Beav < Package
   description 'BEAV (Binary Editor And Viewer), is a full featured binary file editor.'
   homepage 'https://packages.debian.org/sid/beav'
-  version '1.40-18-b3'
+  version '1.40-18-1'
   license 'GPL-1+'
   compatibility 'all'
-  source_url 'file:///dev/null'
-  source_sha256 'e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855'
+  source_url 'https://httpredir.debian.org/debian/pool/main/b/beav/beav_1.40.orig.tar.gz'
+  source_sha256 '7ad905e4124bf105ca0e213d8212bed231559825ac11588794b9be7a91399ddc'
 
-  binary_url ({
-    aarch64: 'https://dl.bintray.com/chromebrew/chromebrew/beav-1.40-18-b3-chromeos-armv7l.tar.xz',
-     armv7l: 'https://dl.bintray.com/chromebrew/chromebrew/beav-1.40-18-b3-chromeos-armv7l.tar.xz',
-       i686: 'https://dl.bintray.com/chromebrew/chromebrew/beav-1.40-18-b3-chromeos-i686.tar.xz',
-     x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/beav-1.40-18-b3-chromeos-x86_64.tar.xz',
-  })
-  binary_sha256 ({
-    aarch64: '61938e9152aa32344551454cc276a46241a83ed1e3332b401b0e950e59963f31',
-     armv7l: '61938e9152aa32344551454cc276a46241a83ed1e3332b401b0e950e59963f31',
-       i686: '703c90a7dcc8a88d731b3d0133b97110c77b96fb05a972bac71774cfac34cdef',
-     x86_64: 'f9f3068e5ac536949578aab4e5009e08cb81e1a0da31a2f51fd65eb37761bf62',
-  })
-
-  depends_on 'alien' => :build
-  depends_on 'libtinfo'
+  def self.patch
+    system "curl -#L https://httpredir.debian.org/debian/pool/main/b/beav/beav_1.40-18.diff.gz | gunzip >> beav_1.40-18.diff"
+    abort 'Checksum mismatch. :/ Try again.'.lightred unless Digest::SHA256.hexdigest( File.read('beav_1.40-18.diff') ) == '5139aa0bdc9424b7ad8197070e1902d8225ee78e2091d02a493422cd91fe5973'
+    system "patch -Np1 -i beav_1.40-18.diff"
+    system "sed -i 's:-lncurses:-ltinfow:g' Makefile"
+    system "sed -i '12d' basic.c"
+    system "sed -i '13d' file.c"
+    system "sed -i '76d' random.c"
+    system "sed -i '11d' symbol.c"
+    system "sed -i 's:#include <term.h>:#include <ncursesw/term.h>:' tcap.c"
```

Comment:
These look like destructive changes.

Reply:
it looks like they are, but I know C, trust me they aren't. This package hasn't been updated since 2004, these are just things that used to be okay, but in newer versions of gcc they've been phased out.